test/NuGet.Core.Tests/NuGet.PackageManagement.Test/BuildIntegratedTests.cs

```diff
@@ -184,144 +184,146 @@ public async Task ChildProjectUpdated_ParentProjectsRestored()
+                buildIntegratedProjects[1].AddProjectReference(buildIntegratedProjects[3]);
+                buildIntegratedProjects[1].AddProjectReference(buildIntegratedProjects[3]);
+                buildIntegratedProjects[1].AddProjectReference(normalProject.PackageSpec);
 
-                    var format = new LockFileFormat();
+                buildIntegratedProjects[2].AddProjectReference(buildIntegratedProjects[3]);
+                buildIntegratedProjects[2].AddProjectReference(normalProject.PackageSpec);
 
-                    // Restore and build cache
-                    var restoreContext = new DependencyGraphCacheContext(logger, testSettings);
+                var message = string.Empty;
+                var format = new LockFileFormat();
 
-                    // Act
-                    await nuGetPackageManager.InstallPackageAsync(buildIntegratedProjects[2], packageIdentity, new ResolutionContext(), new TestNuGetProjectContext(),
-                            sourceRepositoryProvider.GetRepositories(), sourceRepositoryProvider.GetRepositories(), CancellationToken.None);
+                // Restore and build cache
+                var restoreContext = new DependencyGraphCacheContext(logger, testSettings);
 
-                    // Install again
-                    await nuGetPackageManager.InstallPackageAsync(buildIntegratedProjects[2], packageIdentity2, new ResolutionContext(), new TestNuGetProjectContext(),
+                // Act
+                await nuGetPackageManager.InstallPackageAsync(buildIntegratedProjects[2], packageIdentity, new ResolutionContext(), new TestNuGetProjectContext(),
                         sourceRepositoryProvider.GetRepositories(), sourceRepositoryProvider.GetRepositories(), CancellationToken.None);
 
-                    var parsedLockFiles = new List<LockFile>();
-
-                    for (var i = 0; i < 3; i++)
-                    {
-                        var lockFile = format.Read(lockFiles[i]);
-                        parsedLockFiles.Add(lockFile);
-                    }
+                // Install again
+                await nuGetPackageManager.InstallPackageAsync(buildIntegratedProjects[2], packageIdentity2, new ResolutionContext(), new TestNuGetProjectContext(),
+                    sourceRepositoryProvider.GetRepositories(), sourceRepositoryProvider.GetRepositories(), CancellationToken.None);
 
-                    // Assert
-                    Assert.NotNull(parsedLockFiles[0].GetLibrary("NuGet.Configuration", NuGetVersion.Parse("3.3.0")));
-                    Assert.NotNull(parsedLockFiles[1].GetLibrary("NuGet.Configuration", NuGetVersion.Parse("3.3.0")));
-                    Assert.NotNull(parsedLockFiles[2].GetLibrary("NuGet.Configuration", NuGetVersion.Parse("3.3.0")));
-                }
+                // Assert
+                string projectOutputPath0 = buildIntegratedProjects[0].PackageSpec.RestoreMetadata.OutputPath;
```

Comment:
Call GetAssetsFilePathAsync on the projects instead of these 3 lines and the ones below them.